nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -298,7 +496,125 @@ in
         '';
       };
     };
-  };
+
+    services.zfs.zed = {
+      enable = mkOption {
+        default = false;
+        type = types.bool;
+        description = ''
+          Enable the ZFS Event Daemon (ZED).
+        '';
+      };
+
+      debugLog = mkOption {
+        default = null;
+        type = types.nullOr types.path;
+        example = "/tmp/zed.debug.log";
+        description = ''
+          Absolute path to the debug output file.
+        '';
+      };
+ 
+      email.addresses = mkOption {
+        default = [ ];
+        type = types.listOf types.str;
+        example = [ "user@domain.tld" ];        
+        description = ''
+          Email address of the zpool administrator for receipt of notifications.
+          Email will only be sent if value is defined.
+        '';
+      };
+
+      email.program = mkOption {
+        type = types.nullOr types.str;
+        default = "sendmail";
+        example = "mail";
+        description = ''
+          Name or path of executable responsible for sending notifications via email;
+          the mail program must be capable of reading a message body from stdin.
+          Email will only be sent if an email address is defined.
+          By default it will use pkgs.system-sendmail.
+        '';
+      };
+
+      email.options = mkOption {
+        type = types.str;
```

Comment:
`types.listOf types.str` is nicer for a list of commandline arguments.

Reply:
But they need to be in a given order according to the mail program being used. So I think types.listOf types.str is less usefull in this case.

Reply:
Usually command line tools don't care about the order of options, and `listOf str` keeps everything in order anyways, it just concatenates all assignments together. Also it can deal with escaping properly, which might be very important here since this will be used by data received from arbitrary mail subjects and such.